Jenkinsfile

```diff
@@ -22,23 +22,75 @@ properties([
 ])
 
 void setGitHubBuildStatus(String context, String message, String state) {
-  step([
-    $class: 'GitHubCommitStatusSetter',
-    reposSource: [$class: 'ManuallyEnteredRepositorySource', url: 'https://github.com/nuxeo/nuxeo-elements'],
-    contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
-    statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
-  ])
+  if (env.DRY_RUN != "true") {
+    step([
+      $class: 'GitHubCommitStatusSetter',
+      reposSource: [$class: 'ManuallyEnteredRepositorySource', url: 'https://github.com/nuxeo/nuxeo-elements'],
+      contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
+      statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
+    ])
+  }
+}
+
+def isPullRequest() {
+  return BRANCH_NAME =~ /PR-.*/
 }
 
+String getPackageVersion() {
+  container('nodejs') {
+    return sh(script: 'npx -c \'echo "$npm_package_version"\'', returnStdout: true).trim()
+  }
+}
+
+String getReleaseVersion(String preid) {
+  String currentSnapshot = getPackageVersion()
+  String nextPromotion = currentSnapshot.replace("-SNAPSHOT", "")
+  String version = "${nextPromotion}-${preid}.0" // first version ever
+
+  // find the latest tag if any
+  sh "git fetch origin 'refs/tags/v${nextPromotion}*:refs/tags/v${nextPromotion}*'"
+  def tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${nextPromotion}*' | tail -1 | tr -d '\n'")
+  if (tag) {
+    container('nodejs') {
+      version = sh(returnStdout: true, script: "npx semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+String getPullRequestVersion() {
+  return getPackageVersion() + "-${BRANCH_NAME}"
+}
+
+String getVersion() {
+  return isPullRequest() ? getPullRequestVersion() : getReleaseVersion('rc')
```

Comment:
there's really no point in making the `preid` an argument, might as well have it set as a constant and used in `getReleaseVersion()`